File: app/src/main/java/com/example/wanandroidsdk/http/api/ApiService.java
package com.example.wanandroidsdk.http.api;


import com.example.wanandroidsdk.bean.ArticleListBean;
import com.example.wanandroidsdk.bean.EasyDataBean;
import com.example.wanandroidsdk.constant.Constants;
import com.example.wanandroidsdk.http.WanHttpResult;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * @author: njb
 * @date: 2024/3/30 18:04
 * @desc:
 */
public interface ApiService {
    /**
     * 获取首页banner
     */
    @GET(Constants.BANNER_LIST)
    Call<WanHttpResult<List<EasyDataBean>>> banner();

    /**
     * 获取文章列表
     * @param page
     * @return
     */
    @GET(Constants.ARTICLE_LIST)
    Call<WanHttpResult<ArticleListBean>> articleList(@Path("page") int page);

    /**
     *获取搜索热词
     *
     */
    @GET(Constants.HOT_KEY)
    Call<WanHttpResult<List<EasyDataBean>>> getHotKey();

    /**
     * 获取知识
     * @return
     */
    @GET(Constants.TREE)
    Call<WanHttpResult<List<EasyDataBean>>> getKnowLedge();
}

File: app/src/main/java/com/example/wanandroidsdk/http/api/ApiService.kt
package com.example.wanandroidsdk.http.api

import com.example.wanandroidsdk.bean.WanArticleListBean
import com.example.wanandroidsdk.bean.WanDataBean
import com.example.wanandroidsdk.http.WanHttpResult
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET(Constants.BANNER_LIST)
    suspend fun getBanner():  WanHttpResult<List<WanDataBean>>

    @GET(Constants.ARTICLE_LIST)
    suspend fun articleList(@Path("page") page:Int): WanHttpResult<WanArticleListBean>

    @GET(Constants.HOT_KEY)
    suspend fun hotKeyList(): WanHttpResult<List<WanDataBean>>
    
    @GET(Constants.TREE)
    suspend fun knowList(): WanHttpResult<List<WanDataBean>>
}